olive/passes/onnx/conversion.py

```diff
@@ -412,26 +422,53 @@ def _export_ranked_model(params):
             restore_args = replace_tensor_parallel_layers()
 
             input_model = DistributedPyTorchModelHandler(**model_config)
-            olive_pytorch_model = input_model.load_model(local_rank)
-            dummy_inputs = olive_pytorch_model.get_dummy_inputs()
-            io_config = None if pass_config["use_dynamo_exporter"] else olive_pytorch_model.get_io_config()
-            pytorch_model = olive_pytorch_model.prepare_session(rank=local_rank)
 
-            with torch.no_grad():
-                ranked_onnx_model = OnnxConversion._export_pytorch_model(
-                    pytorch_model,
-                    dummy_inputs,
-                    io_config,
-                    pass_config,
-                    device,
-                    torch_dtype,
-                    tempdir,
-                )
+            if input_model.hf_config and input_model.hf_config.components:
+                ranked_models = []
+                for _, component_model in input_model.get_hf_components(local_rank):
+                    dummy_inputs = component_model.get_dummy_inputs()
+                    io_config = None if pass_config["use_dynamo_exporter"] else component_model.get_io_config()
+                    pytorch_model = component_model.prepare_session(rank=local_rank)
+
+                    with torch.no_grad():
```

Comment:
not completely related to this PR, but we could decorate the `_export_pytorch_model` method with `@torch.no_grad()` like in https://github.com/microsoft/Olive/blob/bc8b3312251484986972cf53ac8d4b2f4ed2ca1f/olive/evaluator/olive_evaluator.py#L816. 

This way, it won't matter if we forgot to do the `with torch.no_grad():` like I did in `_convert_model_on_device`

Reply:
Done!